src/core/net/dnssd_server.cpp

```diff
@@ -839,6 +899,62 @@ Error Server::ResolveByQueryCallbacks(Header                 &aResponseHeader,
     return error;
 }
 
+#if OPENTHREAD_CONFIG_DNS_UPSTREAM_QUERY_ENABLE
+void Server::OnUpstreamQueryResponse(UpstreamQueryTransaction &aQueryTransaction, Message &aResponseMessage)
+{
+    Error err = mSocket.SendTo(aResponseMessage, aQueryTransaction.mMessageInfo);
```

Comment:
May be good/safer to check that transaction is valid?
(if platform calls us even if we cancel)